server/mldb_server.cc

```diff
@@ -120,6 +121,15 @@ init(std::string configurationPath,
 
 void
 MldbServer::
+preInit()
+{
+    //Because of a multithread issue in boost, we need to call this to force boost::date_time to initialize in single thread
```

Comment:
Technically speaking, if multiple threads were instantiating an MldbServer simultaneously, the race condition would occur too. One way to guarantee that it does not occur is to call this during the initialisation of the program, possibly via the initialisation of a global variable.